core/controllers/voice_artist.py

```diff
@@ -23,96 +23,115 @@
 from core.domain import fs_domain
 from core.domain import fs_services
 from core.domain import user_services
+from core.domain import voiceover_services
+from core.platform import models
 import feconf
 import python_utils
+import utils
 
 import mutagen
 from mutagen import mp3
 
+(suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
+
+
+def _save_audio_file(
+        raw_audio_file, filename, entity_type, entity_id, user_id):
+    """Saves the given audio file in file system.
+
+    Args:
+        raw_audio_file: *. The raw audio data.
+        filename: str. The filename of the audio.
+        entity_type: str. The type of entity to which the audio belongs.
+        entity_id: str. The id of the entity to which the audio belongs.
+        user_id: str. The ID of the user saving the audio.
+
+    Raises:
+        Exception: If audio not supplied.
+        Exception: If the filename extension is unsupported.
+    """
+    allowed_formats = list(feconf.ACCEPTED_AUDIO_EXTENSIONS.keys())
+
+    if not raw_audio_file:
+        raise Exception('No audio supplied')
+    dot_index = filename.rfind('.')
+    extension = filename[dot_index + 1:].lower()
+
+    if dot_index == -1 or dot_index == 0:
+        raise Exception(
+            'No filename extension: it should have '
```

Comment:
```suggestion
            'No filename extension provided. It should have '
```

Reply:
Done! (This existed earlier in the same way, but I've changed it as suggested.)